spdx-utils/build.gradle

```diff
@@ -59,165 +59,105 @@ def licenseToEnumEntry(String license) {
     return "$enumName(\"$license\")"
 }
 
-task generateSpdxLicenseList {
-    doLast {
-        println 'Fetching SPDX license list...'
+def downloadLicenseTexts(taskName, description, jsonUrl, className, resourcePath, collectIds) {
```

Comment:
Would be nice to have parameter types to prevent errors and make it easier to understand how to call this function. Also the name is a bit misleading because it is not only download texts but also generates an enum, maybe split that into two functions?

I was also thinking about using [KotlinePoet](https://github.com/square/kotlinpoet) for generating the enums, but was not sure if this would make the code shorter or more readable given that the enums have a very simple structure.

Reply:
I've split the code into two functions and added parameter types now.

Reply:
Thanks.